lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3575,6 +3584,24 @@ namespace {
       bool hasReferenceableFields = !members.empty();
       for (auto member : members) {
         auto nd = cast<clang::NamedDecl>(member->getClangDecl());
+
```

Comment:
Is there any way that we could do this before we import the member? Right now we are importing the member and then checking if its parent is invalid, but if for no other reason, I think it would be faster to do it the other way around.

However, I think there may be another reason. If we import this, we are caching it, so that could cause problems, but I'm not 100% sure what they would be right now (I'd have to think about it and play with it a bit).

Reply:
Additionally, do we want to just skip the members, or bail on the type itself? I think there's at least an argument that we don't want to import unions where we can only import some of the members. (And that would make the implementation a lot easier/cleaner.)

Reply:
@zoecarver  Actually the only reason I do it this way is because I am not sure if I will encounter the IndirectFieldDecl before or after the parent CXXRecordDecl in the previous decl/decls loop. I was considering doing a pass to remove the IndirectFieldDecls but thought this would be simpler. What do you think would be a better approach?